src/ng/directive/ngOptions.js

```diff
@@ -492,6 +495,7 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
 
       // We will re-render the option elements if the option values or labels change
       scope.$watchCollection(ngOptions.getWatchables, updateOptions);
+      scope.$watch(attr.ngModel, function() { ngModelCtrl.$render(); }, true);
```

Comment:
The reason we now need this is that we can't assume that the selected model value is not a complex object, in which case the normal `ngModel` checking would fail to notice the change here.